### PR TITLE
Style/router

### LINKED ISSUE #19: Input keyboard
keyboard currently covers inputs. Find solution that displays both

### PR DESCRIPTION
**Pull request by Salamandastron1**
____ Wrote Tests
____ Implemented 
____ Reviewed
# Neccesary checkmarks:
- [] All Tests are Passing

- [x] The code will run locally

## Type of change
- [x] New feature 
- [] Bug Fix

# Implements/Fixes:
* description
closes #19 #14 #2 #1 #4 #5 

# Check the correct boxes

- [x] This broke nothing
- [] This broke some stuff
- [] This broke everything

# Testing Changes
- [x] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have fully tested my code

# Please Include a link to a gif of your feelings about this branch
Link: https://media.giphy.com/media/FoH28ucxZFJZu/giphy.gif

### CODE REVIEW

**Comment by Salamandastron1** [comment]
@Jessica-Erickson If you could look this over then merge. That would be great. Unless of course there is need for changes.

File: .eslintrc.json [hunk 1]
```diff
@@ -8,6 +8,10 @@
       "func-names": 0,
       "implicit-arrow-linebreak": 0,
       "no-use-before-define": ["error", { "variables": false }],
-      "react/prop-types": ["error", { "ignore": ["navigation"] }]
+      "react/prop-types": ["error", { "ignore": ["navigation"] }],
+      "object-curly-newline": 0
+    },
+    "globals": {
+        "fetch": true
```
Comment (SoleraOps) [inline .eslintrc.json:15]: Why is this here?

Comment (Salamandastron1) [reply .eslintrc.json:15]: The linter doesn't recognize "fetch" as a node or JS function. So I have to declare it as such even though react knows what's going on

File: components/CharacterForm.js [hunk 2]
```diff
@@ -1,8 +1,78 @@
...
+      name: '',
+    };
+
+    this.handleInput = this.handleInput.bind(this);
+    this.determineRoute = this.determineRoute.bind(this);
+  }
+
+  handleInput({ nativeEvent }) {
+    this.setState({
+      name: nativeEvent.text,
```
Comment (SoleraOps) [inline components/CharacterForm.js:23]: 'nativeEvent' is both odd and perfect. That is all.

File: components/CharacterForm.js [hunk 3]
```diff
@@ -1,8 +1,78 @@
...
+
+    return (
+      <MainContain>
+        <CardSection>
+          <Input
+            value={name}
+            onChange={this.handleInput}
+            name="name"
+            length={20}
+            placeholder="Character Name"
```
Comment (SoleraOps) [inline components/CharacterForm.js:54]: Should we have the placeholder be an instruction? Or will the format be intuitive enough?

File: components/CodeForm.js [hunk 4]
```diff
@@ -15,27 +18,50 @@ class CodeForm extends Component {
...
+    const { navigation } = this.props;
+    const url = `https://gm-screen-backend.herokuapp.com/api/v1/characters/play_code/${code}`;
+    const playerInfo = await apiCall(url);
+
+    if (playerInfo.created === 1) {
+      navigation.navigate('PlayerScreen', { id: playerInfo.id });
+    } else if (playerInfo.created === 0) {
+      navigation.navigate('CharacterForm', { id: playerInfo.id });
+    } else {
+      Alert.alert('Not a valid code');
```
Comment (SoleraOps) [inline components/CodeForm.js:37]: I'd like to see how this alert works, since alerts tend to be jarring and disruptive, before we keep it.

File: components/CodeForm.js [hunk 5]
```diff
@@ -15,27 +18,50 @@ class CodeForm extends Component {
...
-          Enter Code
-        </Title>
         <CardSection>
           <Input
-            text={code}
+            value={code}
             onChange={this.handleInput}
             name="code"
             length={6}
+            placeholder="Enter Code"
```
Comment (SoleraOps) [inline components/CodeForm.js:52]: For consistency, I think either all placeholders should be instructions or none of them.

File: components/CodeForm.js [hunk 6]
```diff
@@ -15,27 +18,50 @@ class CodeForm extends Component {
...
-  handleInput(e) {
-    console.log(e);
-    this.setState({});
+  handleInput({ nativeEvent }) {
+    this.setState({
+      code: nativeEvent.text,
+    });
+  }
+
+  async determineRoute(code) {
```
Comment (SoleraOps) [inline components/CodeForm.js:27]: This is really well done! Good job!

File: components/Input.js [hunk 7]
```diff
@@ -2,20 +2,24 @@ import React from 'react';
 import { TextInput, View } from 'react-native';
 import PropTypes from 'prop-types';
 
-const Input = ({ text, onChange, name }) => (
+const Input = ({ value, onChange, name, placeholder }) => (
```
Comment (SoleraOps) [inline components/Input.js:5]: Ah, that's where the Input component was coming from.

File: components/PlayerScreen.js [hunk 8]
```diff
@@ -1,8 +1,53 @@
...
 
-const PlayerScreen = () => (
-  <View />
-);
+class PlayerScreen extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      playerName: 'Tuchanka',
```
Comment (SoleraOps) [inline components/PlayerScreen.js:14]: This name sounds/reads dirty/crude. For the demo, can we use something else?

**Review by SoleraOps — APPROVED** [review]

File: components/CodeForm.js (hunk 4 above)
Comment (Salamandastron1) [reply components/CodeForm.js:37]: On phones it's much less jarring. I can show you tomorrow

File: components/CharacterForm.js (hunk 3 above)
Comment (Salamandastron1) [reply components/CharacterForm.js:54]: *shrugs* it's sorta up in the air. Convention is place holder but the title says the same thing

File: components/PlayerScreen.js (hunk 8 above)
Comment (Salamandastron1) [reply components/PlayerScreen.js:14]: It was a placeholder and so it will actually fetch the character name. Fun fact Tuchanka is the krogan home world in Mass effect

File: components/Input.js (hunk 7 above)
Comment (Salamandastron1) [reply components/Input.js:5]: Jess Jess Jess vury sneaky
